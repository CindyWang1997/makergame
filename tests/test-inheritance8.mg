object parent {
  event destroy { std::printstr("parent destroy"); }
}

object son : parent {
  event destroy { std::printstr("son destroy"); }
}

object daughter : parent {
  event destroy { std::printstr("daughter destroy"); super(); super(); }
}

object main {
  event create {
    create son;
    create daughter;
    foreach (parent p) destroy p;
    std::end_game();
  }
}
