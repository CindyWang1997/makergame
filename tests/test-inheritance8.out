son destroy
daughter destroy
parent destroy
parent destroy
